File: 2022/python/main.py
import sys
from src import solutions

if __name__ == "__main__":
    try:
        day = int(sys.argv[1])
    except IndexError:
        day = 0

    if day in [0, 1]:
        print(solutions.day_01.Solution(1).solution())

    if day in [0, 2]:
        print(solutions.day_02.Solution(2).solution())

    if day in [0, 3]:
        print(solutions.day_03.Solution(3).solution())

    if day in [0, 4]:
        print(solutions.day_04.Solution(4).solution())

    if day in [0, 5]:
        print(solutions.day_05.Solution(5).solution())

    if day in [0, 6]:
        print(solutions.day_06.Solution(6).solution())

    if day in [0, 7]:
        print(solutions.day_07.Solution(7).solution())

    if day in [0, 8]:
        print(solutions.day_08.Solution(8).solution())

File: 2022/rust/src/main.rs
#![feature(is_some_and)]
#![feature(if_let_guard)]

use crate::common::solution::Solution;

mod common;
mod solution;

fn main() {
    let day: u8 = std::env::args().nth(1)
        .unwrap_or_else(|| String::from("0"))
        .parse()
        .expect("Failed parsing the day. Example usage: cargo run -- 1");

    if day == 1 || day == 0 { solution::day_01::Day::new().solution(); }
    if day == 2 || day == 0 { solution::day_02::Day::new().solution(); }
    if day == 3 || day == 0 { solution::day_03::Day::new().solution(); }
    if day == 4 || day == 0 { solution::day_04::Day::new().solution(); }
    if day == 5 || day == 0 { solution::day_05::Day::new().solution(); }
    if day == 6 || day == 0 { solution::day_06::Day::new().solution(); }
    if day == 7 || day == 0 { solution::day_07::Day::new().solution(); }

}
